src/renderers/shared/reconciler/__tests__/ReactCompositeComponentError-test.js

```diff
@@ -26,7 +25,9 @@ describe('ReactCompositeComponent-error', function() {
       }
     });
 
+    /*eslint-disable no-new*/
```

Comment:
Changing it to use `void` like the `no-unused-expressions` cases seems to work for this. Would that be preferable? It's hard for me to think of any good places (besides tests) that you'd want to use constructor side-effects.